src/relax/op/tensor/manipulate.cc

```diff
@@ -421,5 +421,285 @@ TVM_REGISTER_OP("relax.flatten")
+    ShapeStructInfo shape_sinfo = Downcast<ShapeStructInfo>(sinfo->shape.value()->struct_info_);
+    if (shape_sinfo->values.defined()) {
+      shape_values.push_back(shape_sinfo->values.value());
+    }
+  }
+
+  if (is_void_dtype) {
+    output_dtype = DataType::Void();
+  }
+  if (output_ndim == kUnknownNDim) {
+    return tensor_sinfo.size() == 1 ? tensor_sinfo[0] : TensorStructInfo(output_dtype, output_ndim);
+  }
+
+  int axis =
+      attrs->axis.defined() ? NormalizeAxis(call, ctx, output_ndim, attrs->axis.value()->value) : 0;
+  // If there is only one input tensor, no action is needed.
+  if (tensor_sinfo.size() == 1) {
+    return tensor_sinfo[0];
+  }
+  if (shape_values.empty()) {
+    return TensorStructInfo(output_dtype, output_ndim);
+  }
+
+  // As long as the there is known shape value, we will do the best effort check to ensure safety.
+  Optional<Array<PrimExpr>> output_shape = CheckComputeOutputShape(call, ctx, shape_values, axis);
+
+  if (shape_unknown || !output_shape.defined()) {
+    return TensorStructInfo(output_dtype, output_ndim);
+  } else {
+    return TensorStructInfo(ShapeExpr(output_shape.value()), output_dtype);
+  }
+}
+
+TVM_REGISTER_OP("relax.concat")
+    .set_attrs_type<ConcatAttrs>()
+    .set_num_inputs(1)
+    .add_argument("data", "Tuple of Tensors", "The input list of tensors.")
+    .set_attr<FInferStructInfo>("FInferStructInfo", InferStructInfoConcat);
+
+/* relax.split */
+TVM_REGISTER_NODE_TYPE(SplitAttrs);
+
+Expr MakeSplit(Expr data, ObjectRef indices_or_sections, int axis) {
+  ObjectPtr<SplitAttrs> attrs = make_object<SplitAttrs>();
+  attrs->indices_or_sections = indices_or_sections;
+  if (!indices_or_sections->IsInstance<ArrayNode>()) {
+    const auto* n_section = indices_or_sections.as<IntImmNode>();
+    CHECK(n_section != nullptr) << "Split op expects the input indices_or_sections to be either an "
+                                   "Array of PrimExpr or an integer. However, the given one is "
+                                << indices_or_sections->GetTypeKey();
+    CHECK_GT(n_section->value, 0) << "Split op expects the input number of sections to be a "
+                                     "positive integer. However, the given number of sections is "
+                                  << n_section->value;
+  }
+  attrs->axis = axis;
+
+  static const Op& op = Op::Get("relax.split");
+  return Call(op, {std::move(data)}, Attrs(attrs), {});
+}
+
+TVM_REGISTER_GLOBAL("relax.op.split").set_body_typed(MakeSplit);
+
+StructInfo InferStructInfoSplit(const Call& call, const BlockBuilder& ctx) {
+  TensorStructInfo data_sinfo = GetUnaryInputTensorStructInfo(call, ctx);
+  const auto* attrs = call->attrs.as<SplitAttrs>();
+  const auto* data_shape = data_sinfo->shape.as<ShapeExprNode>();
+  int axis = data_sinfo->IsUnknownNdim() ? attrs->axis
+                                         : NormalizeAxis(call, ctx, data_sinfo->ndim, attrs->axis);
+
```

Comment:
When data_sinfo is UnnownNdim, consider simply return number n_sections. 

Check what happens if axis is negative but ndim is unknown. What should be the output? It is OK to error in this case, better to be safe

Reply:
There is a long range dep here, add comment about how do we handle such case

Reply:
We only do inference when `data_sinfo->shape` is ShapeExpr, in which case `data_sinfo` must have known ndim and `axis` must have been normalized properly.

I added some comments to bring more clarity.